dbt/include/trino/macros/catalog.sql

```diff
@@ -1,63 +1,127 @@
+        table_name as "table_name",
+        table_type as "table_type",
+        null as "table_owner"
+    from {{ information_schema }}.tables
+{%- endmacro %}
+
+
+{% macro trino__get_catalog_columns_sql(information_schema) -%}
+    select
+        table_catalog as "table_database",
+        table_schema as "table_schema",
+        table_name as "table_name",
+        column_name as "column_name",
+        ordinal_position as "column_index",
+        data_type as "column_type",
+        comment as "column_comment"
+    from {{ information_schema }}.columns
+{%- endmacro %}
+
+
+{% macro trino__get_catalog_table_comment_schemas_sql(information_schema, schemas) -%}
+    {%- for schema in schemas %}
+    select
+        catalog_name as "table_database",
+        schema_name as "table_schema",
+        table_name as "table_name",
+        comment as "table_comment"
+    from system.metadata.table_comments
+    where
+        catalog_name = '{{ information_schema.database }}'
+        and
+        schema_name != 'information_schema'
+        and
+        schema_name = '{{ schema | lower }}'
+    {%- if not loop.last %}
+    union all
+    {% endif -%}
+    {%- endfor -%}
+{%- endmacro %}
 
-        table_comment as (
-            {%- for schema in schemas %}
-            select
-                catalog_name as "table_database",
-                schema_name as "table_schema",
-                table_name as "table_name",
-                comment as "table_comment"
-
-            from system.metadata.table_comments
-            where
-                catalog_name = '{{ information_schema.database }}'
-                and
-                schema_name != 'information_schema'
-                and
-                schema_name = '{{ schema | lower }}'
-            {% if not loop.last %}
-            union all
-            {% endif %}
-            {%- endfor %}
-        )
 
+{% macro trino__get_catalog_table_comment_relations_sql(information_schema, relations) -%}
+    {%- for relation in relations %}
+    select
+        catalog_name as "table_database",
+        schema_name as "table_schema",
+        table_name as "table_name",
+        comment as "table_comment"
+    from system.metadata.table_comments
+    where
+        catalog_name = '{{ information_schema.database }}'
+        and
+        schema_name != 'information_schema'
+        and
+        {% if relation.schema and relation.identifier %}
+                (
+                    schema_name = '{{ relation.schema | lower }}'
+                    and table_name = '{{ relation.identifier | lower }}'
+                )
+            {% elif relation.schema %}
+                (
+                    schema_name = '{{ relation.schema | lower }}'
+                )
+            {% else %}
+                {% do exceptions.raise_compiler_error(
+                    '`get_catalog_relations` requires a list of relations, each with a schema'
+                ) %}
+        {% endif %}
```

Comment:
are you testing all cases here? same for `trino__get_catalog_schemas_where_clause_sql`

Reply:
The only test from `dbt-tests-adapter` is [TestDocsGenerateOverride](https://github.com/dbt-labs/dbt-core/discussions/8307#discussioncomment-7325742). As mentioned in the comments, better tests are needed in `dbt-tests-adapter`, and dbt has committed to prioritizing their creation ([here](https://github.com/dbt-labs/dbt-core/discussions/8307#discussioncomment-7362109)).

Reply:
so are we waiting for these tests before merging this PR?

Reply:
I don't think so, as they probably won't be added in final release `1.7.0`, but rather in some patch release, hopefully in `1.7.1`